lib/testutils/src/lib.rs

```diff
@@ -174,12 +181,14 @@ impl TestEnvironment {
         self.temp_dir.path()
     }
 
+    #[cfg(feature = "git")]
```

Comment:
Is this needed given the one on line 191?

Reply:
I deleted the one on line 191 now.

I'm not sure what's best here. I can make things work by adding `#[cfg(feature = "git")]` before *both* `factories.add_backend` and not having two function definitions. But then I get an annoying clippy that the `mut` in `let mut factories` is unnecessary if clippy is run with `--no-default-features`. It won't show up for most people working on `jj`, but that's the reason I have two function definitions instead.